File: lalxml/swig/octave/check_lalxmlswig.m
## Check SWIG Octave module wrapping lalxml
## Author: Karl Wette, 2011, 2012

## check module load
disp("checking module load ...");
lalxml;
assert(exist("lalxml", "var"));
assert(exist("lalxmlcvar", "var"));
lal;
assert(exist("lal", "var"));
assert(exist("lalcvar", "var"));
disp("PASSED module load");

## check object parent tracking
disp("checking object parent tracking ...");
a = lalxml.new_lalxmlswig_test_parent_map_struct();
for i = 1:7
  b = a.s;
  c = lalxmlcvar.lalxmlswig_test_parent_map.s;
  lalxmlcvar.lalxmlswig_test_parent_map.s = lalcvar.lalswig_test_struct_const;
endfor
clear ans a b c;
CheckMemoryLeaks();
disp("PASSED object parent tracking");

## passed all tests!
disp("PASSED all tests");

File: lalxml/swig/python/check_lalxmlswig.py
# Check SWIG Python module wrapping lalxml
# Author: Karl Wette, 2011, 2012

# check module load
print("checking module load ...")
import lal
import lalxml
from lalxml import cvar as lalxmlcvar
from lal import cvar as lalcvar
print("PASSED module load")

# check object parent tracking
print("checking object parent tracking ...")
a = lalxml.swig_lalxml_test_parent_map_struct()
for i in range(0, 7):
    b = a.s
    c = lalxmlcvar.swig_lalxml_test_parent_map.s
    lalxmlcvar.swig_lalxml_test_parent_map.s = lalcvar.swig_lal_test_struct_const
del a, b, c
lal.CheckMemoryLeaks()
print("PASSED object parent tracking")

# passed all tests!
print("PASSED all tests")
